src/EditorFeatures/Core/Remote/SolutionChecksumUpdater.cs

```diff
@@ -53,9 +44,13 @@ internal sealed class SolutionChecksumUpdater
     private readonly object _gate = new();
     private bool _isSynchronizeWorkspacePaused;
 
+    private readonly IThreadingContext _threadingContext;
+    private readonly CancellationToken _shutdownToken;
+
     public SolutionChecksumUpdater(
         Workspace workspace,
         IAsynchronousOperationListenerProvider listenerProvider,
+        IThreadingContext threadingContext,
         CancellationToken shutdownToken)
```

Comment:
I'm curious if the shutdown token is needed. Instead of just using the disposal token on the threading context

Reply:
I think they are different. VisualStudioWorkspaceServiceHubConnector has StartListening/StopListening methods, which drive the shutdownToken. However, I think the threadingContext's DisposalToken is driven by MEF and I don't think is disposed until VS shutdown.

Reply:
Just saw your other comment about VisualStudioWorkspaceServiceHubConnector. Let me think about that.

Reply:
It looks like right now, VisualStudioWorkspaceServiceHubConnector.StopListening is called on Workspace dispose. If I just send over the threadingContext and use it's cancellation token, are we ok with essentially just using a cancellation token from MEF dispose and not workspace dispose?

Reply:
yeah.  let's keep this as is.  i've looked at hte code and i'm not sure how to reconcile the different Disposes/CTs.  So let's not change that  now.